### PR TITLE
Player Docs: "sUpperBodyIsBusy"

### PR DESCRIPTION
**Pull request by fig02**
- name `sUpperBodyIsBusy`
- add more documentation comment to `Player_UpdateUpperBody`
- change all UpperActionFunctions to return true/false
- rename `upperAnimBlendWeight` to `upperAnimInterpWeight` to correspond more with the "interp" anim task

### CODE REVIEW

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 1]
```diff
@@ -3371,6 +3375,24 @@ int Player_CanUpdateItems(Player* this) {
             (Player_ItemToItemAction(this->heldItemId) == this->heldItemAction));
 }
 
+/**
+ * Updates the Upper Body system.
+ * The Upper Body system is composed of an upper action function and
+ * a seperate skelanime that can play an animation which is different
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_player_actor/z_player.c:3381]:
```suggestion
 * a separate skelanime that can play an animation which is different
```

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 2]
```diff
@@ -3371,6 +3375,24 @@ int Player_CanUpdateItems(Player* this) {
...
+ * from the main skelanime.
+ *
+ * @return true if the upper body is "busy", false otherwise.
+ *
+ * The upper body being "busy" can mean a few things:
+ * - Hookshot has just connected with something that Player can fly to
+ * - A deku is currently being thrown
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_player_actor/z_player.c:3388]: deku nut?

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 3]
```diff
@@ -3371,6 +3375,24 @@ int Player_CanUpdateItems(Player* this) {
...
+ * - The current upper action function has indicated that it is busy
+ *
+ * If an upper action indicates that its busy by returning true, the
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_player_actor/z_player.c:3391]:
```suggestion
 * If an upper action indicates being busy by returning true, the
```

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 4]
```diff
@@ -3371,6 +3375,24 @@ int Player_CanUpdateItems(Player* this) {
...
+ * animation playing in the upper body skeleton will be used.
+ * This animation may play for all limbs or only the upper body limbs
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_player_actor/z_player.c:3393]:
```suggestion
 * This animation may be used for all limbs or only the upper body limbs
```

**Review by Dragorn421 — COMMENTED** [review]
Nice

good job finding such small independent-ish pieces of player

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 5]
```diff
@@ -3371,6 +3375,24 @@ int Player_CanUpdateItems(Player* this) {
...
+/**
+ * Updates the Upper Body system.
+ * The Upper Body system is composed of an upper action function and
+ * a seperate skelanime that can play an animation which is different
+ * from the main skelanime.
+ *
+ * @return true if the upper body is "busy", false otherwise.
+ *
+ * The upper body being "busy" can mean a few things:
+ * - Hookshot has just connected with something that Player can fly to
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_player_actor/z_player.c:3387]: "fly to" sounds kinda weird. Maybe "hooked to" or "moved to"?

Comment (fig02) [reply src/overlays/actors/ovl_player_actor/z_player.c:3387]: original anim name, fwiw: `gPlayerAnim_link_hook_fly_start`
but yeah i would have used the word fly anyway, regardless of their term

**Review by Dragorn421 — APPROVED** [review]

**Review by cadmic — APPROVED** [review]

**Pull request merged**